standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -440,12 +440,13 @@ void findUnknownPartitions(Table table, Set<Path> missingPartDirs,
       partColNames.add(fSchema.getName());
     }
 
+    String defaultPartName = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.DEFAULTPARTITIONNAME);
```

Comment:
**Yes, that case the solution won't work.**
But IMO, its user's responsibility to create the default partition of same type as that of the partition column type and ensure no other datatype partition column is present. 

Consider 2 scenarios where partition column is of **INT** type:
**Scenario1:**
User1 sets default partition conf value to "ANOTHER_PARTITION" and user 2 in another beeline session tries to insert data in default partition, then hdfs dir structure will look something like this:
```
path to table/part=ANOTHER_PARITION
path to table/part=__HIVE_DEFAULT_PARTITION__
```
and running msck repair will never succeed for any user. One user will get this
```
Caused by: java.lang.NumberFormatException: For input string: "ANOTHER_PARITION"
```
and other will get this
```
Caused by: java.lang.NumberFormatException: For input string: "__HIVE_DEFAULT_PARTITION__"
```

**Scenario2:**
User set default partition conf value to INT. Lets say 13, so hdfs dir structure looks like this
```
path to table/part=13
path to table/part=__HIVE_DEFAULT_PARTITION__
```
and in same session user creates a new table pointing of location of old table and run msck repair then it will fail for \_\_HIVE_DEFAULT_PARTITION__.

Unless the user change the config back to \_\_HIVE_DEFAULT_PARTITION__


**Questions that I also have**, is going forward what should be the expected behaviour? That the user has to ensure that there shouldn't be 2 partitions of different datatype than that of partition column as shown in scenario 1?
	
CC @ayushtkn @deniskuzZ  -- please provide your insights on this?



Also, there are multiple places where \_\_HIVE_DEFAULT_PARTITION__ is explicitly used instead of the config value. For instance here:
https://github.com/apache/hive/blob/c27d31722c2a7426f3236d3d892dbe1e206e840d/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/FileUtils.java#L307

Reply:
I just had a quick look seems like the problem is coming post HIVE-23347, which added this method ``getNormalisedPartitionValue`` which does this parsing for numeric values. Before that patch things should have worked & Scenario-1 would have passed as well, with both values being shown in `show partitions`

This method ``getNormalisedPartitionValue`` looks like is valid only for numeric values to sort out some 0s or so. Maybe we can just have a numeric check & return rather than this default partition value matching

```
    if(!NumberUtils.isParsable(partitionValue)) {
      return partitionValue;
    }
```

If we check the regular case without MSCK for scenario-1, the show partitions would show both the values, I tried locally
```
+------------------------------------------+
|                partition                 |
+------------------------------------------+
| month=12/day=2                           |
| month=12/day=3                           |
| month=12/day=__HIVE_DEFAULT_PARTITION__  |
| month=__ANOTHER_PARTITION__/day=9        |
| month=__HIVE_DEFAULT_PARTITION__/day=3   |
+------------------------------------------+
5 rows selected (0.052 seconds)
0: jdbc:hive2://localhost:10000/> 
```

So, post MSCK also we should show both those values. So, I believe we should just check whether the value is numeric or not, if not, just return the value as is..

Reply:
@ayushtkn  Thanks for looking into this. Yes, I have update the PR based on your suggestion